cranelift/filetests/src/test_run.rs

```diff
@@ -34,28 +34,33 @@ fn build_host_isa(
     infer_native_flags: bool,
     flags: settings::Flags,
     isa_flags: Vec<settings::Value>,
-) -> OwnedTargetIsa {
+) -> anyhow::Result<OwnedTargetIsa> {
     let mut builder = cranelift_native::builder_with_options(infer_native_flags)
-        .expect("Unable to build a TargetIsa for the current host");
+        .map_err(|e| anyhow::Error::msg(e))?;
 
     // Copy ISA Flags
     for value in isa_flags {
-        builder.set(value.name, &value.value_string()).unwrap();
+        builder.set(value.name, &value.value_string())?;
     }
 
-    builder.finish(flags).unwrap()
+    let isa = builder.finish(flags)?;
+    Ok(isa)
 }
 
 /// Checks if the host's ISA is compatible with the one requested by the test.
 fn is_isa_compatible(
     file_path: &str,
-    host: &dyn TargetIsa,
+    host: &anyhow::Result<OwnedTargetIsa>,
```

Comment:
This is fairly counterintuitive -- could we take an `Option` or a custom enum here? `Err` turning into some other meaning below seems to go against what I'd expect `Err` to mean.